src/Traits/Rememberable.php

```diff
@@ -21,39 +24,60 @@ public static function bootRememberable()
 
     public static function cacheGet($id)
     {
-        if ($id === null) {
+        return static::cacheGetBy('id', $id);
+    }
+
+    public static function cacheGetBy(string $key, $value)
+    {
+        if ($value === null) {
             return;
         }
 
-        $table = (new static())->getTable();
+        $model = new static();
+
+        throw_unless(
+            in_array($key, $model->remeberableKeys()), Exception::missingKey()
+        );
 
-        if ($model = Cache::get("{$table}:{$id}")) {
+        if ($model = Cache::get("{$model->getTable()}:{$key}:{$value}")) {
             return $model;
         }
 
-        $model = static::find($id);
+        $model = static::firstWhere($key, $id);
 
         return $model ? tap($model)->cachePut() : null;
     }
 
     public function cachePut()
     {
-        $limit = $this->getCacheLifetime();
-        $key = $this->getCacheKey();
+        return $this->remeberableKeys()
+            ->map(fn ($key) => $this->cachePutKey($key))
+            ->last();
+    }
 
-        return $limit === 'forever'
-            ? Cache::forever($key, $this)
-            : Cache::put($key, $this, Carbon::now()->addMinutes($limit));
+    public function getCacheKey($key): string
+    {
+        return "{$this->getTable()}:{$key}:{$this->{$key}}";
     }
 
-    public function getCacheKey()
+    protected function cachePutKey($key)
     {
-        return "{$this->getTable()}:{$this->getKey()}";
+        $limit = $this->getCacheLifetime();
+        $cacheKey = $this->getCacheKey($key);
+
+        return $limit === 'forever'
+            ? Cache::forever($cacheKey, $this)
+            : Cache::put($cacheKey, $this, Carbon::now()->addMinutes($limit));
     }
 
     protected function getCacheLifetime()
     {
         return $this->cacheLifetime
             ?? Config::get('enso.config.cacheLifetime');
     }
+
+    protected function remeberableKeys(): Collection
+    {
+        return Collection::wrap($this->rememberableKeys ?? ['id']);
```

Comment:
maybe we could add a config key here as well?

maybe we should create a `rememberable.php` config and move the lifetime attribute from the enso config?